apps/api/v1/pages/api/event-types/[id]/_patch.ts

```diff
@@ -222,7 +222,7 @@ export async function patchHandler(req: NextApiRequest) {
   };
 
   if (hosts) {
-    await ensureOnlyMembersAsHosts(req, parsedBody);
+    await ensureOnlyMembersAsHosts(req, { hosts, ...parsedBody });
```

Comment:
https://github.com/calcom/cal.com/blob/6104add6bd45c549f9d16170924aaa4f82ef84bb/apps/api/v1/pages/api/event-types/%5Bid%5D/_patch.ts#L209-L216

Host is destructured earlier, so parsedBody.hosts is always defined, because of which this check was never being run. You could add any user as hosts